Tak.py

```diff
@@ -1,21 +1,152 @@
         """Set the initial game state."""
-        board = [board_length][board_length]
-        for i in range(0, board_length):
-            for j in range(0, board_length):
-                board[i][j] = deque()
+        self.board = board
+        if len(self.board) == 0:
+            for i in range(board_length):
+                self.board.append([])
+                for j in range(board_length):
+                    self.board[i].append(deque())
         self.white_pieces = [num_stones, num_capstones]
         self.black_pieces = [num_stones, num_capstones]
         self.placements = [(x, y) for x in range(0, board_length) for y in range(0, board_length)]
         self.white_moves = []
         self.black_moves = []
+        self.board_length = board_length
         moves = self.placements + self.white_moves
-        self.initial = GameState(to_move='W', board=board, pieces=self.white_pieces, moves=moves)
+        self.initial = GameState(to_move='W', board=self.board, pieces=self.white_pieces, moves=moves)
+
+    def terminal_test(self):
+        """Terminal States:\n\t
+        Board Win: all pieces of a player are used or board is covered\n\t
+        Road Win (Dragon): a move results in an orthogonally connected path 
+        of player tiles/capstone pieces belonging to a single player. If a move 
+        causes both players to have roads, the moving player wins."""
+        allPiecesPlaced = self.white_pieces == [0,0] or self.black_pieces == [0,0]
+        
+        fewestPieces = 1
+        #Checks for roads and board coverage
+        for j,row in enumerate(self.board):
+            for i,space in enumerate(row):
+                fewestPieces = min(len(space),fewestPieces) #Tracks board coverage for flat win condition
+                #Road checking begins below
+                if (i == 0 or j == 0):      #We are on the top or left edges of the board
+                    if len(space) > 0:      #There's at least 1 piece in the current location
+                        if space[len(space)-1].type != pieceType.WALL: #Piece can be part of road
+                            if self.findRoads(space[len(space)-1]):
+                                print(f"Road Found! Player: {space[len(space)-1].color}")
+                                return True
+
+        #Score tallies after flat win condition detected
+        if fewestPieces == 1 or allPiecesPlaced:
+            print("Either the board is covered or a player is out of stones! Tallying scores...")
+            white = 0
+            black = 0
+            for row in self.board:
+                for space in row:
+                    if space[len(space)-1].type == pieceType.TILE:
+                        if space[len(space)-1].color == pieceType.WHITE:
+                            white += 1
+                        else:
+                            black += 1
+            print(f"Black: {black} | White: {white}")
```

Comment:
This is a small thing so we don't need to change it, but wouldn't it make more sense to write white first here and in any other location where we have both? Since white moves first I feel like that might be more intuitive.

Reply:
Sure thing. This brings up a good point about including Komi, and weirdly enough reminded me that I don't have any dragon clause logic. Will adjust both in my revisions commit.